spyder/plugins/console/plugin.py

```diff
@@ -4,376 +4,129 @@
-                                    mimedata2url, MENU_SEPARATOR)
-from spyder.plugins.console.widgets.internalshell import InternalShell
-from spyder.widgets.findreplace import FindReplace
-from spyder.plugins.variableexplorer.widgets.collectionseditor import (
-        CollectionsEditor)
-from spyder.widgets.reporterror import SpyderErrorDialog
-from spyder.api.plugins import SpyderPluginWidget
-from spyder.py3compat import to_text_string
+from spyder.api.plugins import SpyderDockablePlugin
+from spyder.api.translations import get_translation
+from spyder.plugins.console.widgets.main_widget import ConsoleWidget
+
 
+# Localization
+_ = get_translation('spyder')
 logger = logging.getLogger(__name__)
 
 
-class Console(SpyderPluginWidget):
+class Console(SpyderDockablePlugin):
     """
     Console widget
     """
-    CONF_SECTION = 'internal_console'
+    NAME = 'internal_console'
+    TABIFY = None
+    REQUIRES = []
+    OPTIONAL = []
+    WIDGET_CLASS = ConsoleWidget
+    CONF_SECTION = NAME
     CONF_FILE = False
-    focus_changed = Signal()
-    redirect_stdio = Signal(bool)
-    edit_goto = Signal(str, int, str)
-    
-    def __init__(self, parent=None, namespace=None, commands=[], message=None,
-                 exitfunc=None, profile=False, multithreaded=False):
-        SpyderPluginWidget.__init__(self, parent)
-        logger.info("Initializing...")
-        self.dialog_manager = DialogManager()
-
-        # Shell
-        self.shell = InternalShell(parent, namespace, commands, message,
-                                   self.get_option('max_line_count'),
-                                   self.get_font(), exitfunc, profile,
-                                   multithreaded)
-        self.shell.status.connect(lambda msg:
-                                  self.sig_show_status_message.emit(msg, 0))
-        self.shell.go_to_error.connect(self.go_to_error)
-        self.shell.focus_changed.connect(lambda: self.focus_changed.emit())
-
-        # Redirecting some signals:
-        self.shell.redirect_stdio.connect(lambda state:
-                                          self.redirect_stdio.emit(state))
+    CONF_FROM_OPTIONS = {
+        'color_theme': ('appearance', 'selected'),
+    }
+
+    # Signals
+    sig_edit_goto_requested = Signal(str, int, str)
+    sig_focus_changed = Signal()
+    sig_refreshed = Signal()
+
+    # --- SpyderDockablePlugin API
+    # ------------------------------------------------------------------------
+    def get_name(self):
+        return _('Internal console')
 
-        # Find/replace widget
-        self.find_widget = FindReplace(self)
-        self.find_widget.set_editor(self.shell)
-        self.find_widget.hide()
-        if parent is not None:
-            self.register_widget_shortcuts(self.find_widget)
-            self.register_widget_shortcuts(self.shell)
+    def get_icon(self):
+        return self.create_icon('stop')  # FIXME:

```

Comment:
Please don't forget to address this.

Reply:
The internal console does not have an icon. I will use `QIcon()` for now